### ISSUE TITLE
th09 ssg functions 

### ISSUE THREAD

**Issue by toimine** [ack7139/thprac#11]
there's a spoileral ssg file for pofv practice, would be cool if thprac could do what it does
[the ssg (both english and japanese versions).zip](https://github.com/ack7139/thprac/files/6020120/the.ssg.both.english.and.japanese.versions.zip)

**Comment by ack7139** [comment]
Just when I'm thinking about enhancing PoFV support. Thanks for providing the SSG files.

**Comment by toimine** [comment]
Np!

**Comment by toimine** [comment]
just realised, it would also be great if one could change rank and boss/charge level (or whatever the 6 and 7 is on https://en.touhouwiki.net/wiki/Phantasmagoria_of_Flower_View/Gameplay, it determines how strong boss attacks and charges are) and maybe have a mode where a charge attack/boss attack is spawned constantly, for l2/l3/boss practice

**Comment by ack7139** [comment]
The "fixed charge attack" one is cool, but (from what I've tested at least) fixing AI's charge attack level makes their bullet dodging ability considerably worse. I'll test further to see if more AI adjustments would help. Rank adjustment will be added for sure.

**Comment by toimine** [comment]
no need for ai adjustments, the ssg i've posted allows adding invincibility to any side of the screen, so you can make ai invincible

**Comment by ack7139** [comment]
Yeah I guess that could be a workaround, but still making AI spam c2/c3 endlessly is not decent enough for a simulation of actual human gameplay. I'll see if I can do better.

**Comment by ack7139** [comment]
Indeed. The usefulness of making AI invincible doesn't restrict to those mentioned above so it will definitely be added.
